src/components/bottom_tabs/tab/index.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import { Animated } from 'react-native'
+
+import * as Styled from './styles'
+import { useTheme } from '@contexts'
+import type { BottomTabBarButtonProps } from '@react-navigation/bottom-tabs'
+
+export interface Props extends BottomTabBarButtonProps {
+  label: string
+  icon: JSX.Element
+}
+
+const Tab = (props: Props): JSX.Element => {
+  const { label, icon, accessibilityState, onPress } = props
+  const { theme } = useTheme()
+
+  const offsetValue = React.useRef(new Animated.Value(100)).current
+  const selected: boolean = accessibilityState.selected
+
+  React.useEffect(() => {
+    if (selected)
+      Animated.spring(offsetValue, {
+        toValue: 0,
+        useNativeDriver: true,
+      }).start()
+    else
+      Animated.spring(offsetValue, {
+        toValue: 100,
+        useNativeDriver: true,
+      }).start()
+  }, [selected, offsetValue])
+
+  return (
+    <Styled.Background
+      onPress={onPress}
+      selected={selected}
+    >
+      <Styled.IconContainer>
+        {React.cloneElement(icon, {
```

Comment:
its possible to create a style under styles.js and call here, or create a component. Check Button, I think its better for readability